src/linux/classes/com/alibaba/wisp/engine/WispEngine.java

```diff
@@ -611,7 +611,7 @@ void finishShutdown() {
                         && task.carrier.engine == WispEngine.this
                         && !task.isThreadTask()
                         && !task.getName().equals(WispTask.SHUTDOWN_TASK_NAME)
-                        && (group == null || task.inDestoryedGroup())) {
+                        && (group == null || task.inDestoryedGroup(group))) {
```

Comment:
`task.inDestoryedGroup()` is more clear. 
Check add a `group == task.group` condition here.